File: CS/E843/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace E843
{
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport
    {
        public XtraReport1()
        {
            InitializeComponent();

            dataSet1.Tables[0].Rows.Add(new Object[]{1, "Williams"});
            dataSet1.Tables[0].Rows.Add(new Object[]{2, "Smith" });
        }

        private void xrLabel1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e)
        {
            ((XRLabel)sender).Tag = GetCurrentColumnValue("ID");
        }

        private void xrLabel1_HtmlItemCreated(object sender, HtmlEventArgs e)
        {
            e.ContentCell.InnerHtml = String.Format("<a href=http://www.testarea.com/property.aspx?id={1}>{0}</a>", e.ContentCell.InnerText, e.Brick.Value);
        }
    }
}

File: VB/E843/XtraReport1.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace E843

    Public Partial Class XtraReport1
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
            dataSet1.Tables(0).Rows.Add(New [Object]() {1, "Williams"})
            dataSet1.Tables(0).Rows.Add(New [Object]() {2, "Smith"})
        End Sub

        Private Sub xrLabel1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs)
            CType(sender, XRLabel).Tag = GetCurrentColumnValue("ID")
        End Sub

        Private Sub xrLabel1_HtmlItemCreated(ByVal sender As Object, ByVal e As HtmlEventArgs)
            e.ContentCell.InnerHtml = [String].Format("<a href=http://www.testarea.com/property.aspx?id={1}>{0}</a>", e.ContentCell.InnerText, e.Brick.Value)
        End Sub
    End Class
End Namespace
